presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -1257,6 +1258,36 @@ public void testLambdaWithSubqueryInOrderBy()
     }
 
     @Test
+    public void testLambdaWithInvalidParameterCount()
+    {
+        assertFails(INVALID_PARAMETER_USAGE, "line 1:27: Expected a lambda that takes 1 arguments but got 2", "SELECT apply(5, (x, y) -> 6)");
+        assertFails(INVALID_PARAMETER_USAGE, "line 1:30: Expected a lambda that takes 1 arguments but got 3", "SELECT apply(5, (x, y, z) -> 6)");
```

Comment:
the error location is lambda _body_ instead of lambda declaration (so in CLI `6` gets marked red instead of whole `(x, y, z) -> 6`). Would it be possible to report location better?